java/code/src/com/suse/manager/webui/utils/gson/PackageStateJson.java

```diff
@@ -164,8 +168,7 @@ public String getArch() {
                     VersionConstraints vc = versionConstraint.get();
                     if (!Arrays.asList(VersionConstraints.LATEST, VersionConstraints.ANY)
                             .contains(vc)) {
-                        packageState.setEvr(PackageEvrFactory.lookupOrCreatePackageEvr(
-                                getEpoch(), getVersion(), getRelease()));
+                        packageState.setEvr(packageEvr);
```

Comment:
Is it sufficient to use packageEvr directly? It might not contain the DB ID when it was just initialized with `new`.
Do we need the ID?

Reply:
This is now fixed in the reworked version going back to using the lookup function and taking the evr type as parameter to this function instead.